postgres/macros.jinja

```diff
@@ -25,7 +25,7 @@
 {{ state }}-{{ name }}:
   {{ state }}.{{ ensure|default('present') }}:
     {{- format_kwargs(kwarg) }}
-    - onchanges:
```

Comment:
Instead of making the requirement, if I'd suggest to put a conditional like this:
```jinja
    {%- if needs_client_binaries %}
    - onchanges:
       - test: postgres-reload-modules
    {%- endif %}
```

This would make changes only when the binaries were not available before, which is valid behavior.
Otherwise, there would not be any dependency on the test state, since it would actually do nothing.